teuthology/orchestra/remote.py

```diff
@@ -449,6 +449,43 @@ def get_tar_stream(self, path, sudo=False):
             ])
         return self.run(args=args, wait=False, stdout=run.PIPE)
 
+    def write_file(self, path, data, sudo=False, mode=None, owner=None,
+                                     mkdir=False, append=False):
+        """
+        Write data to remote file
+
+        :param path:    file path on remote host
+        :param data:    str, binary or fileobj to be written
+        :param sudo:    use sudo to write file, defaults False
+        :param mode:    set file mode bits if provided
+        :param owner:   set file owner if provided
+        :param mkdir:   preliminary create the file directory, defaults False
+        :param append:  append data to the file, defaults False
+        """
+        dd = 'sudo dd' if sudo else 'dd'
+        args = dd + ' of=' + path
+        if append:
+            args += ' conv=notrunc oflag=append'
+        if mkdir:
+            mkdirp = 'sudo mkdir -p' if sudo else 'mkdir -p'
+            dirpath = os.path.dirname(path)
+            if dirpath:
+                args = mkdirp + ' ' + dirpath + '\n' + args
+        if mode:
+            chmod = 'sudo chmod' if sudo else 'chmod'
+            args += '\n' + chmod + ' ' + mode + ' ' + path
+        if owner:
+            chown = 'sudo chown' if sudo else 'chown'
+            args += '\n' + chown + ' ' + owner + ' ' + path
+        args = 'set -ex' + '\n' + args
+        self.run(args=args, stdin=data)
+
+    def sudo_write_file(self, path, data, **kwargs):
```

Comment:
IMHO, `sudo_write_file()` is not super-convenient compared to `write_file(path, data,  sudo=True)`. So I won't mind not having `sudo_write_file()`.

But, yes, not having it would break many tests.

Reply:
I can delete it, I don't it can break anything because it is new function, I added because someone might think it handy to use `sudo_write_file` because of shorter syntax.

Reply:
> But, yes, not having it would break many tests.

this is why we do not delete `teuthology.misc.sudo_write_file` but make it use `remote.write_file(sudo=True)`

Reply:
i'd like to see PRs on ceph side which use the newly added functions. otherwise these new methods are just unused and untested.

Reply:
@tchaikov it is already used there, through misc.sudo_write_file, check there logs:
https://pulpito.ceph.com/kyr-2020-07-16_20:09:04-rados:cephadm-master-distro-basic-smithi/
You can find there a 'dd' command.
Even if you don't see it used in upstream ceph, it does not mean we (mostly me) do not plan to use it downstream.
I agree we can add more test cases, however it is "tested" through misc api.

I don't want to refactor and go through the PRs on ceph side now and go through the all cycles of merging and backporting. 
I'm more comfortable to do this later, this is why a deprication message.

Reply:
> I don't want to refactor and go through the PRs on ceph side now and go through the all cycles of merging and backporting.
> I'm more comfortable to do this later, this is why a deprication message.

please take your time. we can have all of them once they are ready. 

[OT] also, please let me know if you are going to work on https://github.com/ceph/ceph/pull/35674#discussion_r442871795 ? if not, i will cherry-pick the fixes.

Reply:
@tchaikov what do you have on mind, to backport it to 'octopus', 'nautilus', or also py2 depended branches 'luminous' and correspondingly backport this PR to teuthology py2?

Reply:
@tchaikov is that kind of PR do you expect? https://github.com/ceph/ceph/pull/36211

Reply:
> [OT] also, please let me know if you are going to work on [ceph/ceph#35674 (comment)](https://github.com/ceph/ceph/pull/35674#discussion_r442871795) ? if not, i will cherry-pick the fixes.

Do you suggest to finish it before merging this PR?

Reply:
> > [OT] also, please let me know if you are going to work on [ceph/ceph#35674 (comment)](https://github.com/ceph/ceph/pull/35674#discussion_r442871795) ? if not, i will cherry-pick the fixes.
> 
> Do you suggest to finish it before merging this PR?

no, it's note related to this changeset. i marked "[OT]" (short for off-topic) here. as per your reply, i just realized that you are not likely to work on backports. hence the question.

Reply:
> no, it's not related to this changeset. i marked "[OT]" (short for off-topic) here. as per your reply, i just realized that you are not likely to work on backports. hence the question.

thanks for the remark, sorry sometimes I'm bad to decode things like "ot". Regarding backport, sooner or later I might work on it. But not right now. So go ahead and cherry-pick it.